### PR TITLE
Problem 25

### PR DESCRIPTION
**Pull request by karlkeefer**
Here's a pull request with working code for [problem 25](https://projecteuler.net/problem=25) of project euler.
You can see the code changes if you look at the "files changed" tab of this PR, and even leave comments directly inline.

I posted some comments in the code to help explain what's going on.

### CODE REVIEW

**Comment by karlkeefer** [comment]
It's worth noting that there are non-brute-force solutions to this, but you have to be a huge math nerd to figure them out.

You can checkout this code on your machine by doing `git checkout problem-25` (after cloning the repo), and then using nodejs to run the script with `node main.js`.

This is all very command-line-y and might be super frustrating to figure out without us pairing even for just a few minutes to get your local machine setup properly.

**Comment by karlkeefer** [comment]
@niallgiblin were you able to see the diff on the files changed tab?

Now that it's merged, you can either:
1) clone the whole repo which will now include these changes on the `master` branch or 
2) run `git pull` if you've already cloned the repo, and it will pull changes from github to your local machine

Then, you can try to actually run my code to get the answer for problem 25, though if you don't already have nodejs running on your machine that will be its own challenge 😂

File: 25/main.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+// use BigInt because javascript numbers don't go high enough
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER
+let a = BigInt(1);
+let b = BigInt(1);
+
+// i is our index, it starts at 2 because we've already set a and b to initial values
+let i = 2;
+
+// loop until we hit a 1000-digit number
+while (b.toString().length < 1000) {
```
Comment (karlkeefer) [inline 25/main.js:11]: @niallgiblin I don't have anything important to say, just showing you that you can make comments on specific lines of code in the Pull Request!

**Comment by niallgiblin** [comment]
> 

Yep I saw them! I'm having trouble cloning the repo though, not entirely sure why. I do have nodejs running

**Comment by karlkeefer** [comment]
The _best way_ to clone is using the ssh-style git connections.
e.g. `git clone git@github.com:karlkeefer/euler.git`

but for those to work you have to generate an ssh key and upload it to github (or you might _already have_ an ssh key to upload).

Strong recommend the github docs for troubleshooting that: https://docs.github.com/en/authentication/connecting-to-github-with-ssh

**Comment by karlkeefer** [comment]
`ssh` is not-at-all beginner friendly, so don't spend too much effort on this before kicking it to a pair session where I can show you the ropes ;)

**Pull request merged**